src/HLint.hs

```diff
@@ -153,8 +153,8 @@ runHints args settings cmd@CmdMain{..} = do
     j <- if cmdThreads == 0 then getNumProcessors else pure cmdThreads
     withNumCapabilities j $ do
         let outStrLn = whenNormal . putStrLn
-        ideas <- getIdeas cmd settings
-        ideas <- pure $ if cmdShowAll then ideas else  filter (\i -> ideaSeverity i /= Ignore) ideas
+        ideas' <- getIdeas cmd settings
+        let ideas = if cmdShowAll then ideas' else  filter (\i -> ideaSeverity i /= Ignore) ideas'
```

Comment:
If you do want to do this, making it `(if cmdShowAll then id else filter ...) <$> getIdeas cmd settings` looks better.

Reply:
I don't like using `fmap` or `<$>` on the RHS of `<-` for basically the same reason that I don't like `<- pure` (see my other comment below).

Reply:
I feel like you probably won't like this, but how about `(if cmdShowAll then id else filter ... -> ideas) <- getIdeas cmd`?